Fix missing __error symbol in libbtrfs.so.0.

The easiest way to reproduce it is to try to build
btrfs-progs with LDFLAGS=-Wl,--no-undefined :

btrfs-list.o: In function `lookup_ino_path':
btrfs-list.c:(.text+0x7d2): undefined reference to `__error'

Reported-by: Denis Descheneaux
Bug: https://bugs.gentoo.org/show_bug.cgi?id=613890
diff --git a/Makefile b/Makefile
index 67fbc48..05a5d06 100644
--- a/Makefile
+++ b/Makefile
@@ -97,3 +97,3 @@ objects = ctree.o disk-io.o kernel-lib/radix-tree.o extent-tree.o print-tree.o \
 	  inode.o file.o find-root.o free-space-tree.o help.o send-dump.o \
-	  fsfeatures.o messages.o
+	  fsfeatures.o
 cmds_objects = cmds-subvolume.o cmds-filesystem.o cmds-device.o cmds-scrub.o \
@@ -106,3 +106,3 @@ cmds_objects = cmds-subvolume.o cmds-filesystem.o cmds-device.o cmds-scrub.o \
 libbtrfs_objects = send-stream.o send-utils.o kernel-lib/rbtree.o btrfs-list.o \
-		   kernel-lib/crc32c.o \
+		   kernel-lib/crc32c.o messages.o \
 		   uuid-tree.o utils-lib.o rbtree-utils.o
@@ -448,3 +448,3 @@ test-ioctl: ioctl-test ioctl-test-32 ioctl-test-64
 
-library-test: library-test.o messages.o $(libs_shared)
+library-test: library-test.o $(libs_shared)
 	@echo "    [LD]     $@"
@@ -454,3 +454,3 @@ library-test: library-test.o messages.o $(libs_shared)
 
-library-test.static: library-test.static.o messages.static.o $(libs_static)
+library-test.static: library-test.static.o $(libs_static)
 	@echo "    [LD]     $@"
